pkg/controller/gitopsservice/gitopsservice_controller.go

```diff
@@ -160,8 +162,28 @@ func (r *ReconcileGitopsService) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{}, err
 	}
 
+	defaultArgoCDInstance, err := argoCDCR(request.Name, request.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Set GitopsService instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, defaultArgoCDInstance, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	existingArgoCD := &argoapp.ArgoCD{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: defaultArgoCDInstance.Name, Namespace: defaultArgoCDInstance.Namespace}, existingArgoCD)
```

Comment:
There are no tests for this code.

Reply:
Yes, wanted to let folks know about the approach before writing tests. Will add them today :)